src/test/java/teammates/test/cases/action/InstructorCourseRemindActionTest.java

```diff
@@ -171,39 +171,34 @@ public void testExecuteAndPostProcess() throws Exception {
                 Const.ParamsNames.INSTRUCTOR_EMAIL, invalidEmail
         };
         
-        try {
-            remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
-            redirectResult = (RedirectResult) remindAction.executeAndPostProcess();
-            signalFailureToDetectException();
-        } catch (EntityNotFoundException e) {
-            ignoreExpectedException();
-        }
+        testExecuteAndPostProcess_assertEntityNotFoundException(instructorId, submissionParams);
         
         submissionParams = new String[]{
                 Const.ParamsNames.COURSE_ID, courseId,
                 Const.ParamsNames.STUDENT_EMAIL, invalidEmail
         };
         
-        try {
-            remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
-            redirectResult = (RedirectResult) remindAction.executeAndPostProcess();
-            signalFailureToDetectException();
-        } catch (EntityNotFoundException e) {
-            ignoreExpectedException();
-        }
-        
+        testExecuteAndPostProcess_assertEntityNotFoundException(instructorId, submissionParams);
+
+        ______TS("Failure case: Invalid course id parameter");
+
         submissionParams = new String[]{
-                Const.ParamsNames.COURSE_ID, "invalidCourseId"
+                Const.ParamsNames.COURSE_ID, "invalidCourseId",
+                Const.ParamsNames.INSTRUCTOR_EMAIL, anotherInstructorOfCourse1.email
         };
         
+        testExecuteAndPostProcess_assertEntityNotFoundException(instructorId, submissionParams);
+    }
+
+    private void testExecuteAndPostProcess_assertEntityNotFoundException(String instructorId,
+                                                                         String[] submissionParams) {
         try {
-            remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
-            redirectResult = (RedirectResult) remindAction.executeAndPostProcess();
-            signalFailureToDetectException();
+            Action remindAction = getAction(addUserIdToParams(instructorId, submissionParams));
```

Comment:
Instead of `Action`, use the specific subclass here, i.e. `InstructorFeedbackRemindAction`.